src/main/java/com/example/icecreamup/dao/ProductDAO.java

```diff
@@ -1,22 +1,30 @@
 package com.example.icecreamup.dao;
 
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
 import com.example.icecreamup.dto.ProductDTO;
 
+/**
+ * 
+ * @author chaklahk
+ */
+@Component
 public class ProductDAO implements IProductDAO {
 
+	@Autowired
 	ProductRepository productRepository;
 	
 	@Override
-	public boolean save(ProductDTO productDTO) throws Exception {
-		// TODO Auto-generated method stub
+	public boolean save(ProductDTO productDTO) throws Exception
+	{

```

Comment:
The methods were already in place to implement these in the same way as OrderDAO